controller/controller.go

```diff
@@ -237,9 +245,80 @@ func (c *Controller) deleteFunc(ctx context.Context, obj interface{}) {
 	}
 }
 
+// hashEvent takes an event and produces an integer hash that:
+// - is guaranteed to be stable for the lifetime of an object,
+// - is in the bounds 0 <= int < c.concurrency.
+func (c *Controller) hashEvent(ctx context.Context, event watch.Event) int {
+	v := reflect.ValueOf(event.Object).Elem()
```

Comment:
Not deep enough into apimachinery to find a better way to do this. General problem is to gets some stable ID from `runtime.Object` to use as input to the hash function. Probably better ways to do it.